[CI] CsvTests test {inlinestats.ShadowingSelf} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3520 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/larwjwc2ox4nk)
- [elasticsearch-pull-request #26915 / part-3](https://gradle-enterprise.elastic.co/s/r5itrcamy6lp4)
- [elasticsearch-periodic-platform-support #3520 / ubuntu-2004-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/otnvepwqsa4xs)
- [elasticsearch-periodic #3546 / graalvm-ce17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/3vzdl26m2fr5m)
- [elasticsearch-periodic-platform-support #3516 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/5vqd2uqdhncyq)
- [elasticsearch-periodic-platform-support #3516 / windows-2016_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/ycqvjbl2djniq)
- [elasticsearch-periodic-platform-support #3516 / ubuntu-2004-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/whhf3w3mo5i4o)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.CsvTests" -Dtests.method="test {inlinestats.ShadowingSelf}" -Dtests.seed=89957FD1CA267EC5 -Dtests.locale=tr -Dtests.timezone=Europe/Volgograd -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Binlinestats.ShadowingSelf%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.CsvTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <0L>
     but: was <32L>
```

**Issue Reasons:**
- [main] 2 consecutive failures in step ubuntu-2004-aarch64_checkpart3_platform-support-arm
- [main] 7 failures in test test {inlinestats.ShadowingSelf} (9.9% fail rate in 71 executions)
- [main] 2 failures in step ubuntu-2004-aarch64_checkpart3_platform-support-arm (100.0% fail rate in 2 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

I've reproduced with:
```
while ./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.CsvTests" -Dtests.method="test {inlinestats.*" -Dtests.iters=100; do echo ok; done
```

Looks like:
```
java.lang.AssertionError: 
Expected: <0L>
     but: was <176L>
	at __randomizedtesting.SeedInfo.seed([165CD9EFB39AE1F1:7591BAC4E316D2D8]:0)
	at org.elasticsearch.xpack.esql.CsvTests.inlinestats.Shadowing(inlinestats.csv-spec:403)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2449)
	at org.elasticsearch.xpack.esql.CsvTests.doTest(CsvTests.java:289)
	at org.elasticsearch.xpack.esql.CsvTests.test(CsvTests.java:263)
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -117,9 +117,6 @@ tests:
 - class: org.elasticsearch.action.admin.indices.create.SplitIndexIT
   method: testSplitIndexPrimaryTerm
   issue: https://github.com/elastic/elasticsearch/issues/111282
-- class: org.elasticsearch.xpack.esql.CsvTests
-  method: test {inlinestats.ShadowingSelf}
-  issue: https://github.com/elastic/elasticsearch/issues/111261
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java
@@ -286,7 +286,8 @@ public class CsvTests extends ESTestCase {
             assertWarnings(actualResults.responseHeaders().getOrDefault("Warning", List.of()));
         } finally {
             Releasables.close(() -> Iterators.map(actualResults.pages().iterator(), p -> p::releaseBlocks));
-            assertThat(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST).getUsed(), equalTo(0L));
+            // Give the breaker service some time to clear in case we got results before the rest of the driver had cleaned up
+            assertBusy(() -> assertThat(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST).getUsed(), equalTo(0L)));
         }
     }
 

